### PR TITLE
Add multithreading tests for 5 validation-check tasks (Group 4)

### PR DESCRIPTION
**Pull request by SimaTian**
Squashed merge of 5 multithreading test additions for validation-check tasks.

Stacked on #52909 (multithreading-polyfills).

## Tasks (test-only, no source changes needed for 4 of 5)
- CheckForDuplicateFrameworkReferences
- CheckForImplicitPackageReferenceOverrides
- CheckForUnsupportedWinMDReferences
- CheckIfPackageReferenceShouldBeFrameworkReference
- CollatePackageDownloads

## Changes (6 files)
5 new multithreading stress test files + 1 source change:
- \\CheckForUnsupportedWinMDReferences.cs\\: adds \\[MSBuildMultiThreadableTask]\\ attribute (was missing on base branch despite the merge prompt noting otherwise)

These tasks already had \\IMultiThreadableTask\\ on the polyfills branch. This commit adds multithreading unit tests to verify thread safety.

## Test Results
5 passed, 0 failed (multithreading tests only).
Build and full test suite succeed.

### CODE REVIEW

**Review by SimaTian — COMMENTED** [review]
Addressed review comments:

1. **Unused using**: Removed `using Microsoft.Build.Framework` from TaskEnvironmentHelperTests.cs.
2. **TestDriverProxy**: Changed from `public` to `internal`.
3. **Blank lines**: Added missing blank lines in GivenACollatePackageDownloadsMultiThreading.cs.
4. **PR description scope**: This PR targets `multithreading-polyfills`, so diff against main includes polyfill files from base PR (#52909). The actual changes are test files + one task source change.

Branch rebased on updated polyfills and force-pushed.

File: src/Tasks/Common/AbsolutePath.cs [hunk 1]
```diff
@@ -125,7 +125,7 @@ public AbsolutePath(string path, AbsolutePath basePath)
                 throw new ArgumentException("Path must not be null or empty.", nameof(path));
             }
 
-            Value = Path.Combine(basePath.Value, path);
+            Value = Path.GetFullPath(Path.Combine(basePath.Value, path));
```
Comment (JanProvaznik) [inline src/Tasks/Common/AbsolutePath.cs]: no

**Review by JanProvaznik — CHANGES_REQUESTED** [review]
I think concurrency tests are a bad idea, if static analysis identified no static fields and no process state usage then that seems sufficient

the edge case tests seem nice addition to coverage

not sure about the conventions of sdk repo but test class per task would be much nicer than cramming them to 2 disorganized files

the output parity is weird, i'd prefer assert to expected result which is precomputed and then do a theory with both taskenvironments

why are there some tests that just run the task and assert it succeds, isn't that duplicate to what should be in some testing already?

File: src/Tasks/Microsoft.NET.Build.Tasks/CheckForUnsupportedWinMDReferences.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@ namespace Microsoft.NET.Build.Tasks
 {
     //  This task is used for projects targeting .NET 5 and higher and generates errors if there are any
     //  unsupported WinMD references.
+    [MSBuildMultiThreadableTask]
```
Comment (JanProvaznik) [inline src/Tasks/Microsoft.NET.Build.Tasks/CheckForUnsupportedWinMDReferences.cs]: remove this, the task is not migrated, see line 53 AssemblyHasWindowsRuntimeReference resolves path

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/TaskEnvironmentHelper.cs [hunk 3]
```diff
@@ -74,7 +73,7 @@ public static TaskEnvironment CreateForTest(string projectDirectory)
     /// Stores its own project directory independently from the process's CWD,
     /// enabling tests to verify tasks resolve paths relative to ProjectDirectory, not CWD.
     /// </summary>
-    public class TestDriverProxy : DispatchProxy
+    internal class TestDriverProxy : DispatchProxy
```
Comment (JanProvaznik) [inline src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/TaskEnvironmentHelper.cs:77]: ?

**Review by JanProvaznik — APPROVED** [review]
looks ok, increases coverage

⚠️  revert the extra marking of task that sneaked in
